public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -237,110 +272,141 @@ const Policy: FC<PolicyComponentProps> = ({
+
+              return (
+                <Policy
+                  key={uniqueId()}
+                  routeTree={routeTree}
+                  currentRoute={child}
+                  receivers={receivers}
+                  contactPointsState={contactPointsState}
+                  readOnly={isThisChildReadOnly}
+                  inheritedProperties={childInheritedProperties}
+                  onAddPolicy={onAddPolicy}
+                  openDetailModal={openDetailModal}
+                  onDeletePolicy={onDeletePolicy}
+                  onShowAlertInstances={onShowAlertInstances}
+                  alertManagerSourceName={alertManagerSourceName}
+                  alertGroups={alertGroups}
+                  routesMatchingFilters={routesMatchingFilters}
+                  matchingInstancesPreview={matchingInstancesPreview}
+                  isAutoGenerated={isThisChildAutoGenerated}
+                  isAutogeneratedChunkOpen={isAutogeneratedChunkOpen}
+                  toggleAutogeneratedChunk={toggleAutogeneratedChunk}
+                />
+              );
+            })}
+          </>
+        </div>
+        {showExportDrawer && <GrafanaPoliciesExporter onClose={toggleShowExportDrawer} />}
+      </Stack>
+    </ConditionalWrap>
+  ) : null;
+};
 
-          return (
-            <Policy
-              key={uniqueId()}
-              routeTree={routeTree}
-              currentRoute={child}
-              receivers={receivers}
-              contactPointsState={contactPointsState}
-              readOnly={readOnly || provisioned}
-              inheritedProperties={childInheritedProperties}
-              onAddPolicy={onAddPolicy}
-              onEditPolicy={onEditPolicy}
-              onDeletePolicy={onDeletePolicy}
-              onShowAlertInstances={onShowAlertInstances}
-              alertManagerSourceName={alertManagerSourceName}
-              alertGroups={alertGroups}
-              routesMatchingFilters={routesMatchingFilters}
-              matchingInstancesPreview={matchingInstancesPreview}
-            />
-          );
-        })}
-      </div>
-      {showExportDrawer && <GrafanaPoliciesExporter onClose={toggleShowExportDrawer} />}
-    </Stack>
+export const AUTOGENERATED_ROOT_LABEL_NAME = '__grafana_autogenerated__';
+
+export function isAutoGeneratedRootAndSimplifiedEnabled(route: RouteWithID) {
+  const simplifiedRoutingToggleEnabled = config.featureToggles.alertingSimplifiedRouting ?? false;
+  if (!simplifiedRoutingToggleEnabled) {
+    return false;
+  }
+  if (!route.object_matchers) {
+    return false;
+  }
+  return (
+    route.object_matchers.some((objectMatcher) => {
+      return (
+        objectMatcher[0] === AUTOGENERATED_ROOT_LABEL_NAME &&
+        objectMatcher[1] === MatcherOperator.equal &&
+        objectMatcher[2] === 'true'
+      );
+    }) ?? false
   );
-};
+  //  return route.receiver === 'contact_point_5';
```

Comment:
Remove?

Reply:
I let this comment for testing purposes...once we integrate the BE changes we can remove it